tools/spelling/common.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import re
+from dataclasses import dataclass
+
+
+CWD = os.getcwd()
+CSPELL_DOCKER = "ghcr.io/streetsidesoftware/cspell:latest"
+CSPELL_COMMAND = f"docker run -it -v {CWD}:/workdir {CSPELL_DOCKER} \
+lint --no-exit-code --config tools/spelling/cspell.json"
+FIXME_FILE = "fixme.spelling"
+DICT_FILE = "tools/spelling/cspell_dicts/cspell_dict.txt"
```

Comment:
I am biased a bit, but I liked the separation of usernames, layouts and other names into their own dictionaries. I guess it would be hard to keep it up with the automation though :(

Reply:
That would be nice, but rather impossible for a simple tool :(